core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1038,9 +1038,8 @@ class DAGScheduler(
     }
 
     if (tasks.size > 0) {
-      logInfo("Submitting " + tasks.size + " missing tasks from " + stage + " (" + stage.rdd + ")")
-      stage.pendingPartitions ++= tasks.map(_.partitionId)
-      logDebug("New pending partitions: " + stage.pendingPartitions)
+      logInfo(s"Submitting ${tasks.size} missing tasks (for partitions " +
+        s"${tasks.map(_.partitionId)}) from $stage (${stage.rdd})")
```

Comment:
I think including all the partitions could make this log line a little too long for an info message (what if there are 1k or 10k tasks).  I understand not wanting to stick it in a debug message (you often dont' realize you want debug logging on until *after* something has failed), but I think this will be too much for regular runs.

Would it still be useful if you truncate to the first 10 tasks?  `tasks.take(10).map(...)`